sktime/datatypes/_hierarchical/_check.py

```diff
@@ -0,0 +1,127 @@
+
+Parameters
+----------
+obj - object to check
+return_metadata - bool, optional, default=False
+    if False, returns only "valid" return
+    if True, returns all three return objects
+var_name: str, optional, default="obj" - name of input in error messages
+
+Returns
+-------
+valid: bool - whether obj is a valid object of mtype/scitype
+msg: str - error message if object is not valid, otherwise None
+        returned only if return_metadata is True
+metadata: dict - metadata about obj if valid, otherwise None
+        returned only if return_metadata is True
+    fields:
+        "is_univariate": bool, True iff all series in hier.panel have one variable
+        "is_equally_spaced": bool, True iff all series indices are equally spaced
+        "is_equal_length": bool, True iff all series in panel are of equal length
+        "is_empty": bool, True iff one or more of the series in the panel are empty
+        "is_one_series": bool, True iff there is only one series in the hier.panel
+        "is_one_panel": bool, True iff there is only one flat panel in the hier.panel
+        "has_nans": bool, True iff the panel contains NaN values
+        "n_instances": int, number of instances in the hierarchical panel
+        "n_panels": int, number of flat panels in the hierarchical panel
+"""
+
+__author__ = ["fkiraly"]
+
+__all__ = ["check_dict"]
+
+import numpy as np
+import pandas as pd
+
+from sktime.datatypes._series._check import check_pdDataFrame_Series
+
+VALID_INDEX_TYPES = (pd.Int64Index, pd.RangeIndex, pd.PeriodIndex, pd.DatetimeIndex)
+VALID_MULTIINDEX_TYPES = (pd.Int64Index, pd.RangeIndex)
+
+
+def _list_all_equal(obj):
+    """Check whether elements of list are all equal.
+
+    Parameters
+    ----------
+    obj: list - assumed, not checked
+
+    Returns
+    -------
+    bool, True if elements of obj are all equal
+    """
+    if len(obj) < 2:
+        return True
+
+    return np.all([s == obj[0] for s in obj])
+
+
+check_dict = dict()
+
+
+def _ret(valid, msg, metadata, return_metadata):
+    if return_metadata:
+        return valid, msg, metadata
+    else:
+        return valid
+
+
+def check_pdmultiindex_Hierarchical(obj, return_metadata=False, var_name="obj"):
+
+    if not isinstance(obj, pd.DataFrame):
+        msg = f"{var_name} must be a pd.DataFrame, found {type(obj)}"
+        return _ret(False, msg, None, return_metadata)
+
+    if not isinstance(obj.index, pd.MultiIndex):
+        msg = f"{var_name} must have a MultiIndex, found {type(obj.index)}"
+        return _ret(False, msg, None, return_metadata)
+
+    nlevels = obj.index.nlevels
+    if not nlevels > 2:
+        msg = f"{var_name} have a MultiIndex with 3 or more levels, found {nlevels}"
+        return _ret(False, msg, None, return_metadata)
+
+    inst_inds = obj.index.droplevel(-1).unique()
+    panel_inds = inst_inds.droplevel(-1).unique()
+
+    check_res = [
+        check_pdDataFrame_Series(obj.loc[i], return_metadata=True) for i in inst_inds
```

Comment:
Where is this check_pdDataFrame_Series please burn it and rename it properly to: check_pd_dataframe_series

Reply:
this is not added by the PR, we can do this in a separate PR - let's not widen scope, or I have to go through multiple files.
I can carry out a lower_snake_case renaming later.